viewer/packages/camera-manager/src/StationaryCameraManager.ts

```diff
@@ -181,7 +206,40 @@ export class StationaryCameraManager implements CameraManager {
     this._camera.quaternion.setFromEuler(euler);
 
     this._cameraChangedListeners.forEach(cb => cb(this._camera.position, this.getTarget()));
-  };
+  }
+
+  private pinchCamera(moveEvent: PointerEvent) {
+    if (this._downEventCache.length < 2) {
+      return;
+    }
+
+    const getEuclideanDistance = (eventOne: PointerEvent, eventTwo: PointerEvent): number => {
+      const dx = eventOne.clientX - eventTwo.clientX,
+        dy = eventOne.clientY - eventTwo.clientY;
+      return Math.sqrt(dx * dx + dy * dy);
+    };
+
+    const preMoveDistance = getEuclideanDistance(this._downEventCache[0], this._downEventCache[1]);
+    const downEventIndex = this._downEventCache.findIndex(event => event.pointerId === moveEvent.pointerId);
+    const preMoveEvent = this._downEventCache[downEventIndex];
+    this._downEventCache[downEventIndex] = moveEvent;
+    const postMoveDistance = getEuclideanDistance(this._downEventCache[0], this._downEventCache[1]);
+    const distanceDelta = postMoveDistance - preMoveDistance;
+
+    // To stop FOV stuttering we only update if the
+    // change in distance is above a small threshold.
+    const sensitivityThreshold = 1.0;
+    if (Math.abs(distanceDelta) < sensitivityThreshold) {
+      this._downEventCache[downEventIndex] = preMoveEvent;
+      return;
+    }
+
+    const { width, height } = this._domElement.getBoundingClientRect();
+    const screenSize = Math.sqrt(width * width + height * height);
+    if (screenSize > 0) {
+      this.setFOV(this._camera.fov - (distanceDelta * 100) / screenSize);
```

Comment:
Pull out the 100 in a constant with a descriptive name.

Reply:
This is just normal percentage calculation, but I pulled the entire calculation out to make this clear.